registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java

```diff
@@ -520,16 +527,36 @@ private boolean validateRegistrationID(SyncRegistrationDto registrationDto,
 			syncRegistration.setWorkflowInstanceId(RegistrationUtility.generateId());
 			syncRegistrationDao.save(syncRegistration);
 			syncResponseDto.setRegistrationId(registrationDto.getRegistrationId());
-
+			
 			eventId = EventId.RPR_407.toString();
 		}
 		syncResponseDto.setStatus(ResponseStatusCode.SUCCESS.toString());
 		syncResponseList.add(syncResponseDto);
+		saveAnonymousProfile( registrationDto,  referenceId,  timeStamp);
 		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
 				registrationDto.getRegistrationId(), "SyncRegistrationServiceImpl::validateRegId()::exit");
 		return syncResponseList;
 	}
 
+	private void saveAnonymousProfile(SyncRegistrationDto registrationDto, String referenceId, String timeStamp)  {
+		AnonymousProfileDTO dto=new AnonymousProfileDTO();
+		try{
+			dto.setProcessName(registrationDto.getRegistrationType());
+			dto.setStatus("REGISTERED");
+			dto.setStartDateTime(timeStamp);
+			dto.setDate(LocalDate.now(ZoneId.of("UTC")).toString());
+			dto.setProcessStage("SYNC");
+			dto.setEmail(dto.getEmail() != null ? getHashCode(dto.getEmail()) : null);
+			dto.setEnrollmentCenterId(referenceId.split("_")[0]);
+			dto.setPhone(dto.getPhone() != null ? getHashCode(dto.getPhone()) : null);

```

Comment:
please dont set phone or email actual value. Its like this -
1. if phone is there then from mapping json read phone field. Update the field here. If phone number is not there then update as null
2. same for email.

Reply:
I have changed the code